bundles/model/org.eclipse.smarthome.model.script/src/org/eclipse/smarthome/model/script/interpreter/ScriptInterpreter.xtend

```diff
@@ -77,11 +79,12 @@ public class ScriptInterpreter extends XbaseInterpreter {
 
     override protected invokeFeature(JvmIdentifiableElement feature, XAbstractFeatureCall featureCall,
         Object receiverObj, IEvaluationContext context, CancelIndicator indicator) {
-        if (feature != null && feature.eIsProxy) {
+        if (feature !== null && feature.eIsProxy) {
             if (featureCall instanceof XMemberFeatureCall) {
                 throw new ScriptExecutionException(new ScriptError(
                     "'" + featureCall.getConcreteSyntaxFeatureName() + "' is not a member of '" +
-                        receiverObj?.getClass()?.getName() + "'", featureCall));
+                        (((featureCall?.memberCallTarget as XMemberFeatureCall)?.feature as JvmOperation)?.returnType as JvmParameterizedTypeReference)?.type.qualifiedName
```

Comment:
While this returns exactly what we want in the current case, I'm not sure if this is the right implementation. The problem I see here is that it makes a couple of assumptions (i.e. memberCallTarget is a XAbstractFeatureCall, its feature is a JvmOpration, returnType is a JvmTypeReference) and to be honest I am not sure if this will always be the case. Sounds unlikely to me. I'll see if I can dig up something better...

To be honest, I

Reply:
Just confirmed: The following now _miserably_ fails:

```xtend
  logger.info("simon says", new StringType().toUnit("°F"));
```

with 

```
15:35:18.004 ERROR o.e.s.m.r.r.i.e.ExecuteRuleJob[:69] - Error during the execution of rule 'Initialize sjka items': An error occurred during the script execution: org.eclipse.xtext.xbase.impl.XConstructorCallImplCustom cannot be cast to org.eclipse.xtext.xbase.XAbstractFeatureCall
```

although it actually is the same problem and therefore _should_ behave identically from the user's perspective.

Reply:
I think I found one:

```xtend

    @Inject
    private IBatchTypeResolver typeResolver;

    // ...

                val expression = featureCall.memberCallTarget;
                val type = typeResolver.resolveTypes(expression)?.getActualType(expression)?.identifier;
                throw new ScriptExecutionException(new ScriptError(
                    "'" + featureCall.getConcreteSyntaxFeatureName() + "' is not a member of '" + type + "'", featureCall));

```

Reply:
Thanks a lot @sjka, I will update the PR.